src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py

```diff
@@ -293,3 +309,229 @@ def upload(self, url, args, filename):
+                       'PNN': None,
+                       'GUID': None
+                       }
+
+        msg = "Writing DQM root files to CERN EOS with retries: %s and retry pause: %s"
+        logging.info(msg, eosmanager.numberOfRetries, eosmanager.retryPauseTime)
+        try:
+            eosmanager(eosFileInfo)
+        except Alarm:
+            msg = "Indefinite hangout while staging out to EOS"
+            logging.error(msg)
+            raise WMExecutionFailure(70317, "DQMUploadStageOutTimeout", msg)
+        except Exception as ex:
+            msg = "EOS copy failed, lfn: %s. Error: %s" % (eosFileInfo['LFN'], str(ex))
+            logging.exception(msg)
+            raise WMExecutionFailure(70318, "DQMUploadFailure", msg)
+
+        return eosFileInfo['LFN']
+
+    def getAnalysisFileLFN(self, analysisFile):
+        """
+        Construct an LFN for an analysisFile
+        """
+        # If lfn start with '/', make it relative to it
+        lfnFile = os.path.relpath(analysisFile.lfn, start='/')
+        lfn = os.path.join(self.registerLFNBase, lfnFile)
+    
+        return lfn
+
+    def _register(self, registerURL, args):
+        """
+        POST request to register URL
+        """
+
+        ident = "WMAgent python/%d.%d.%d" % sys.version_info[:3]
+        uploadProxy = self.step.upload.proxy or os.environ.get('X509_USER_PROXY', None)
+        logging.info("Using proxy file: %s", uploadProxy)
+        logging.info("Using CA certificate path: %s", os.environ.get('X509_CERT_DIR'))
+
+        msg = "HTTP Register POST arguments: %s\n" % args
+        logging.info(msg)
+
+        handler = HTTPSAuthHandler(key=uploadProxy, cert=uploadProxy)
+        opener = OpenerDirector()
+        opener.add_handler(handler)
+
+        # setup the request object
+        url = decodeBytesToUnicode(registerURL)
+        datareq = Request(url)
+        datareq.add_header('Accept-encoding', 'gzip')
+        datareq.add_header('User-agent', ident)
+        datareq.data = encodeUnicodeToBytes(json.dumps(args))
+
+        if 'https://' in url:
+            result = opener.open(datareq)
+        else:
+            opener.add_handler(ProxyHandler({}))
+            result = opener.open(datareq)
+
+        return result
+    
+    def _registerAnalysisFile(self, stepName, analysisFile):
+        """
+        Register a file to new DQM GUi
+        - https://github.com/cms-DQM/dqmgui
+        HTTP request body:
+            [{"dataset": "/a/b/c", "run": "123456", "lumi": "0", "file": "/eos/cms/store/group/comm_dqm/DQMGUI_data/location/file.root", "fileformat": 1}]
+        Set lumis to 0, as this is reproducing current per Run based root files
+        - https://github.com/dmwm/WMCore/issues/10287#issuecomment-1052558061
+        DQM root files produced by Harvesting prosessing are type 1 (plain ROOT)
+        - https://github.com/cms-DQM/dqmgui#file-formats
+        """
+        # Get task description
+        job = Bootstrap.loadJobDefinition()
```

Comment:
Kenyi, could you please check whether the job description is already available in this object? From:
https://github.com/dmwm/WMCore/blob/ee8c9fe76301997917e938b998393760893de7cf/src/python/WMCore/WMSpec/Steps/Executor.py#L93

it might be that we already have it under `self.job`

Reply:
Good catch! I will check